sips/sip-031/sip-031.md

```diff
@@ -0,0 +1,552 @@
+
+### Technical Implementation
+A new Epoch 3.2 will manage the emission of the new tokens and its schedule. To facilitate mints and transfers, Epoch 3.2 relies on a boot contract exposing:
+
+* A public function update-principal that allows setting the current-principal. The principal set in this smart contract will elect the recipient of the initial 100M STX (minted at epoch transition), the recipient of the unlockings for the second 100M STX over 24 months, and the recipient of the 300M STX transferred over 60 months (at each tenure). 
+* A read-only function get-principal returning the current principal (set with update-principal)
+* A read-only function get-original-principal, returning the principal that received the mint of 100M STX (locked and unlocking over 24 months).
+* A public function, unlock-at-bitcoin-height, that given a bitcoin height returns non-zero if any amount unlocks (over the 24 months phase). 
+
+```clarity
+(define-public (update-principal (new-principal principal) (response bool uint))
+(define-read-only (get-principal) (response principal uint))
+(define-read-only (get-original-principal) (response principal uint))
+(define-public (unlock-at-bitcoin-height (bitcoin-height uint) (response (tuple (original principal) (recipient principal)) uint)))
+```
+
+This design allows the principal that will receive funds during the 5 years cycle to change (e.g., in case of a rotation in the treasury committee), while keeping the smart contract extremely simple to reduce potential attack surface and implementation complexity.
+
+* When instantiated on mainnet, current-principal will be a 3 out of 5 multisig, deterministically generated starting from the public keys of the 5 treasury signers. 
+* To rotate the signing set, current-principal will issue a call to update-principal.
+* All mint events of epoch 3.2 will produce a receipt that will be attached to the coinbase of the block in which the mint happened.
+
+#### 100M STX at epoch transition
+At epoch transition (activation), 100M STX will be minted and transferred to the recipient resulting from the get-principal call of contract above. The mint will produce a corresponding receipt (attached to the coinbase transaction as described above).
+
+#### 100M STX unlocking in 24 iterations
+At epoch transition (activation), 100M STX will be minted and transferred to the recipient resulting from the get-principal call of contract, where they will remain locked. 
```

Comment:
Should this be transferred to the contract, not to the principal specified in the contract? If I'm understanding the intentions here, then the STX should be locked in the contract until distributed following the details below, every ~month.